Sources/Auth/Services/Common/NetworkingInteractor.swift

```diff
@@ -1,18 +1,40 @@
 import Foundation
 import WalletConnectRelay
+import WalletConnectUtils
+import WalletConnectKMS
 
 protocol NetworkInteracting {
     func subscribe(topic: String) async throws
+    func request(_ request: JSONRPCRequest<AuthRequestParams>, topic: String, envelopeType: Envelope.EnvelopeType) async throws
 }
 
+extension NetworkInteracting {
+    func request(_ request: JSONRPCRequest<AuthRequestParams>, topic: String, envelopeType: Envelope.EnvelopeType = .type0) async throws {
+        try await self.request(request, topic: topic, envelopeType: envelopeType)
+    }
+}
+
+
 class NetworkingInteractor: NetworkInteracting {
     private let relayClient: RelayClient
+    private let serializer: Serializing
+    private let jsonRpcHistory: JsonRpcHistory<AuthRequestParams>
```

Comment:
As it was already merged to develop, what do you also think of using the improved `RPCHistory` in here, so you can avoid binding the storage to a single specialized type?